### PR TITLE
CORE-948 Pipelines how it works

### PR DESCRIPTION
**Pull request by ellisonc**
Add docs for how it works page in pipelines section

### CODE REVIEW

File: _docs-sources/pipelines/how-it-works/index.md [hunk 1]
```diff
@@ -1,68 +1,81 @@
...
+inside Gruntwork Pipelines. We have [example configurations](https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/master/examples/for-production/infrastructure-live/_ci/scripts)
+that identify changed terraform modules and call the Gruntwork Pipelines invoker lambda.
 
-## Invoker Lambda
+By default, the invoker lambda is run by a CLI tool called `infrastructure-deployer` from within your CI tool.
 
-Oh my goodness, have I told you about the time I saw a beautiful Blue Jay perched on my windowsill? It was truly a sight to behold! But I digress... let me tell you about AWS Lambda functions, which are like the busy little hummingbirds of the cloud computing world.
+## ECS Deploy Runner
 
-Picture this: a Lambda function is like a bird feeder that sits and waits for requests to come in, just like a bird feeder waits for feathered friends to come and snack. But instead of seed, it's programmed to receive code and execute it.
```
Comment (zackproser) [inline _docs-sources/pipelines/how-it-works/index.md:17]: wtf :laughing:

File: _docs-sources/pipelines/how-it-works/index.md [hunk 2]
```diff
@@ -1,68 +1,81 @@
...
-Kaniko was a humble tool, born of the desire to provide a secure and reliable way to build Docker images in a Kubernetes environment. It did not seek glory or power, but instead sought to serve the people, to make their lives easier and their applications more secure.
 
-And Kaniko did work diligently, using its many talents to build Docker images in a way that was both efficient and trustworthy. It did not rely on a Docker daemon, but instead used a container to perform the build process, ensuring that the image was built in a clean and isolated environment.
+### Invoker Lambda
+
+The [Invoker Lambda](https://github.com/gruntwork-io/terraform-aws-ci/blob/main/modules/ecs-deploy-runner/main_lambda.tf)
+is a lambda function written in python that acts as the AWS entrypoint for your pipeline.
+It has 3 primary roles:
+
+1. It serves as a gatekeeper for pipelines runs, determining if a particular command is allowed to be run
```
Comment (zackproser) [inline _docs-sources/pipelines/how-it-works/index.md:53]:
```suggestion
1. It serves as a gatekeeper for pipelines runs, determining if a particular command is allowed to be run, and if the arguments passed to that command are valid
```

File: docs/pipelines/how-it-works/index.md [hunk 3]
```diff
@@ -1,76 +1,89 @@
...
+The Docker Image Builder task definition allows CI jobs to build docker images.
+This ECS task uses an open source library called [Kaniko](https://github.com/GoogleContainerTools/kaniko) to enable docker builds from within a docker container.
+We provide a [docker image](https://github.com/gruntwork-io/terraform-aws-ci/tree/main/modules/ecs-deploy-runner/docker/kaniko) based on kaniko for this task.
 
-The ECS Deploy Runner is like a fierce direwolf, swift and agile, capable of executing complex deployment tasks with ease. It is the tool of choice for those who seek to deploy their applications with speed and precision.
+#### Packer AMI Builder
 
-The Terraform Planner is like a wise and ancient dragon, possessing an almost infinite knowledge of the infrastructure it seeks to conquer. With its keen intellect, it can survey the land and plan the deployment with utmost accuracy, leaving no stone unturned in its quest for victory.
+The Packer AMI Builder task definition allows CI jobs to build AMIs using Hashicorp's Packer tool. This task runs in
```
Comment (zackproser) [inline docs/pipelines/how-it-works/index.md:74]:
```suggestion
The Packer AMI Builder task definition allows CI jobs to build AMIs using HashiCorp's Packer tool. This task runs in
```

Comment (zackproser) [reply docs/pipelines/how-it-works/index.md:74]: Super nit

File: _docs-sources/pipelines/how-it-works/index.md [hunk 4]
```diff
@@ -1,68 +1,81 @@
...
+The Docker Image Builder task definition allows CI jobs to build docker images.
+This ECS task uses an open source library called [Kaniko](https://github.com/GoogleContainerTools/kaniko) to enable docker builds from within a docker container.
+We provide a [docker image](https://github.com/gruntwork-io/terraform-aws-ci/tree/main/modules/ecs-deploy-runner/docker/kaniko) based on kaniko for this task.
 
-The ECS Deploy Runner is like a fierce direwolf, swift and agile, capable of executing complex deployment tasks with ease. It is the tool of choice for those who seek to deploy their applications with speed and precision.
+#### Packer AMI Builder
 
-The Terraform Planner is like a wise and ancient dragon, possessing an almost infinite knowledge of the infrastructure it seeks to conquer. With its keen intellect, it can survey the land and plan the deployment with utmost accuracy, leaving no stone unturned in its quest for victory.
+The Packer AMI Builder task definition allows CI jobs to build AMIs using Hashicorp's Packer tool. This task runs in
```
Comment (MoonMoon1919) [inline _docs-sources/pipelines/how-it-works/index.md:74]:
```suggestion
The Packer AMI Builder task definition allows CI jobs to build AMIs using Hashicorp Packer. This task runs in
```

**Comment by zackproser** [comment]
Looking good locally - I'm not sure if the second right navigation blurb/rail on the how-it-works page is currently adding any value. To wit, I think it may have been partially responsible for a weird scrolling issue I just had when testing it out.

File: _docs-sources/pipelines/how-it-works/index.md [hunk 5]
```diff
@@ -1,68 +1,81 @@
...
+a [docker image](https://github.com/gruntwork-io/terraform-aws-ci/tree/main/modules/ecs-deploy-runner/docker/deploy-runner) we provide.
 
-The Applier, on the other hand, is like a steadfast and loyal knight, resolute in its determination to carry out the orders of its master. With unflinching obedience, it applies the deployment plan, ensuring that every detail is executed with precision and care.
+#### Terraform Planner/Applier
```
Comment (MoonMoon1919) [inline _docs-sources/pipelines/how-it-works/index.md:77]: The title makes this read like one configuration and confused my brain when i was reading the intro paragraph of this section where it states there are 4 . Maybe `Terraform Planner and Terraform Applier`?

File: docs/pipelines/how-it-works/index.md (hunk 3 above)
Comment (ellisonc) [reply docs/pipelines/how-it-works/index.md:74]: My hobby is triggering the people who create branding guidelines 😉

**Review by MoonMoon1919 — COMMENTED** [review]
Looks great! Just a few tiny nits

File: _docs-sources/pipelines/how-it-works/index.md (hunk 1 above)
Comment (oredavids) [reply _docs-sources/pipelines/how-it-works/index.md:17]: 😂

File: _docs-sources/pipelines/how-it-works/index.md [hunk 6]
```diff
@@ -1,68 +1,81 @@
...
-Where containers reign with power and pride.
-This runner, like a loyal squire true,
-Doth serve those who would deploy anew.
+Usage:
 
-With ECS as its liege, it doth obey,
-And to the cloud, it doth make its way.
-In rolling updates, it doth excel,
-Ensuring no downtime, all shall be well.
+`infrastructure-deployer --aws-region AWS_REGION [other otpions] -- CONTAINER_NAME SCRIPT ARGS...`
```
Comment (arsci) [inline _docs-sources/pipelines/how-it-works/index.md:31]:
```suggestion
`infrastructure-deployer --aws-region AWS_REGION [other options] -- CONTAINER_NAME SCRIPT ARGS...`
```

**Pull request closed without merging**